src/app/submission/sections/sections.service.ts

```diff
@@ -367,12 +405,21 @@ export class SectionsService {
    *    The section id
    * @param data
    *    The section data
-   * @param errors
-   *    The list of section errors
+   * @param errorsToShow
+   *    the list of the section's errors to show
+   * @param serverValidationErrors
+   *    the list of the section errors detected by the server
```

Comment:
I'm not sure I understand the difference between these two parameters.  Could we describe (in these comments) why two separate lists of errors are necessary? Are there scenarios where we do NOT want to show an error detected by the server? 

It's just not currently clear to me when these lists would be *different*, but it seems like there must be scenarios where they are different.

Reply:
I've improved comments hoping is more clear now